### PR TITLE
Refine the init function

### PR DESCRIPTION
**Pull request by yhe39**
To support different case.

Tracked-On: OAM-104363
Signed-off-by: HeYue <yue.he@intel.com>

### CODE REVIEW

**Comment by yhe39** [comment]
hi @YuanjunHuang  @feijiang1 please help to review.
This patch support different use cases of android.
i verify on:
CIV: BM/GVT-d/VirtIO/SR-IOV boot well
IVI: BM/GVT-d/VirtIOV/SR-IOV/iGPU+dGPU boot well, the SR-IOV+dGPU i contact ACRN team, they return the device so do not have the env and couldn't help to try now. will contact them help to try SR-IOV + dGPU when they get device.

File: drm/ResourceManager.cpp [hunk 1]
```diff
@@ -71,6 +72,40 @@ void ResourceManager::Init() {
...
+          drms_.emplace_back(std::move(dev));
+        }
+      } else {
+        if (dev) {
+          drms_.emplace_back(std::move(dev));
+        }
+      }
+    }
+    // is SRI-IOV + dGPU
+    if (node_num == 3) {
```
Comment (feijiang1) [inline drm/ResourceManager.cpp:106]: suggest add more comments, also there is typo:
is SR-IOV + dGPU case, Celadon only use card2(virtio-gpu) for display

File: drm/ResourceManager.cpp [hunk 2]
```diff
@@ -71,6 +72,40 @@ void ResourceManager::Init() {
...
+    // only have one card, is BM/GVT-d/Virtio
+    if (node_num == 1) {
+      std::ostringstream path;
+      path << path_pattern << 0;
+      auto dev = DrmDevice::CreateInstance(path.str(), this);
+      if (dev) {
+        drms_.emplace_back(std::move(dev));
+      }
+    }
+    // is SR-IOV or iGPU + dGPU
```
Comment (feijiang1) [inline drm/ResourceManager.cpp:87]: add more comments, like:
For SR-IOV case, use virtio-gpu(card1) for display
For iGPU+dGPU case, only use iGPU(card0) for display

**Review by feijiang1 — COMMENTED** [review]
Thanks Yue, just need add more comments in code.

**Comment by yhe39** [comment]
hi @feijiang1, has modify accordingly. please help to review.
Thanks!

**Review by feijiang1 — APPROVED** [review]

**Pull request merged**